spec/examples/record/record_set_spec.rb

```diff
@@ -811,6 +811,46 @@ def self.latest(count)
     end
   end
 
+  describe '#page_size' do
+    it 'should perform query with specified page_size' do
+      expect_query_with_options(/SELECT/, :page_size => 1) do
+        Post.page_size(1).to_a
+      end
+    end
+
+    it 'should return the number of records specified by page_size' do
+      expect(Post.page_size(2).to_a.length).to be(2)
+    end
+  end
+
+  describe '#next_paging_state' do
+    it 'should return the paging state of the result' do
+      expect(Post.page_size(1).next_paging_state).not_to eq(nil)
+    end
+  end
+
+  describe '#paging_state' do
+    let(:page_one) { Post.page_size(1) }
+    let(:page_two) { Post.page_size(1).paging_state(page_one.next_paging_state) }
+
+    it 'should perform query with specified paging_state' do
+      expect(page_one.next_paging_state).not_to eq(nil)
+      expect_query_with_options(/SELECT/, :paging_state => page_one.next_paging_state) do
+        page_two.to_a
+      end
+    end
+
+    it 'should return results starting after previous page' do
+      expect_query_with_options(/SELECT/, :paging_state => page_one.next_paging_state) do
+        expect(page_two.to_a.first).to eq(posts.to_a.second)
+      end
+    end
+
+    it 'should be nil when omitted' do
+      expect(Post.page_size(1).paging_state(nil).to_a).to eq(page_one.to_a)
+    end
+  end
+
```

Comment:
Sure.  I'm not heavily wed to the tests as they are.  Its probably obvious I don't have much experience with library development, I think I was trying to do too many things at once.  

I wanted some independence to the tests, but really the page_size/paging_state/next_paging_state are very codependent.  I was also trying to fit in with how I saw the consistency tests being done since the page_size and paging_state follow similar paths.  Really the primary thing is to demonstrate that the use of the page_size/paging_state are effective at paginating.  

It sounds like that's more of what you're looking for: use the page_size/paging_state to page through the records(say, in a loop), validate that the pages are of the correct size, and that the collected pages are the correct set.  And it wouldn't do any of the 'expect_query_with_options' checks.  Sound about right?